ghostferry_db.go

```diff
@@ -0,0 +1,44 @@
+package ghostferry
+
+import (
+	"context"
+	"database/sql"
+)
+
+type GhostferryDB struct {
+	SqlDB *sql.DB
+}
+
+type GhostferryTx struct {
+	Tx *sql.Tx
```

Comment:
Can you embed the pointer directly as opposed having a member variable? This should reduce the amount of `tx.Tx`.